File: DotReactiveCocoa/DotReactiveCocoaDemo/DotReactiveCocoaDemo/ViewController.m
//
//  ViewController.m
//  DotReactiveCocoaDemo
//
//  Created by Khoa Pham on 4/11/15.
//  Copyright (c) 2015 Fantageek. All rights reserved.
//

#import "ViewController.h"
#import "RACSignal+FTGDot.h"
#import <ReactiveCocoa/ReactiveCocoa.h>

@interface ViewController ()

@property (nonatomic, copy) NSString *username;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    RACObserve(self, username)
    .ftg_filter(^(NSString *newName){
        return [newName hasPrefix:@"j"];
    })
    .ftg_subscribeNext(^(NSString *newName){
        NSLog(@"%@", newName);
    });

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Signal/Example/Signal/ViewController.swift
//
//  ViewController.swift
//  Signal
//
//  Created by Khoa Pham on 12/30/2015.
//  Copyright (c) 2015 Khoa Pham. All rights reserved.
//

import UIKit
import Signal

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let s = Signal<String>()

        s.subscribeNext { value in
            print(value)
        }

        s.sendNext("Happy New Year")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

